crates/phactory/src/lib.rs

```diff
@@ -399,12 +404,31 @@ impl<Platform: pal::Platform> Phactory<Platform> {
 impl<P: pal::Platform> Phactory<P> {
     // Restored from checkpoint
     pub fn on_restored(&mut self) -> Result<()> {
+        self.check_requirements();
         self.reconfigure_network();
         if let Some(system) = &mut self.system {
             system.on_restored()?;
         }
         Ok(())
     }
+
+    fn check_requirements(&self) {
```

Comment:
Does it check whenever the chain has updated the min version?

Reply:
We do the check at each block after the chain state has been validated and before dispatching the events.
Or after a checkpoint has been loaded.

Reply:
OK